### PR TITLE
Files Table Template Command Line

### PR DESCRIPTION
**Pull request by jhoanmanuelms**
- Create command line tool for the files table template transpiler
- Process the files contained in the provided path
- Fix bugs in the transpiler
- Improve unit test coverage

### CODE REVIEW

File: commons/commons-transpiler/src/test/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLineTest.kt [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        assertThat(pageTab).isEqualTo("pageTab")
+        verify(exactly = 0) { mockHelpFormatter.printHelp(any(), any()) }
+    }
+
+    @Test
+    fun `missing arguments`() {
+        val pageTab = testInstance.transpile(arrayOf("-f", "JSON"))
+
+        assertThat(pageTab).isEqualTo("")
+        verify(exactly = 1) { mockHelpFormatter.printHelp(any(), any()) }
```
Comment (Juan-EBI) [inline commons/commons-transpiler/src/test/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLineTest.kt]: personally unless not possible I prefer to assert mocks with specific values as `any` may hide a bug in the code

File: commons/commons-transpiler/src/main/kotlin/ac/uk/ebi/transpiler/exception/InvalidColumnException.kt [hunk 2]
```diff
@@ -0,0 +1,8 @@
+package ac.uk.ebi.transpiler.exception
+
+const val INVALID_COLUMN_ERROR_MSG = "Base columns must match the template"
+
+class InvalidColumnException : RuntimeException() {
```
Comment (Juan-EBI) [inline commons/commons-transpiler/src/main/kotlin/ac/uk/ebi/transpiler/exception/InvalidColumnException.kt]: I would include the column that generate the problem in the error message so problems is easier to identify

File: commons/commons-transpiler/src/main/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLine.kt [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+            pageTab = transpiler.transpile(
+                template = template.readText(),
+                baseColumns = columns.split(",").map { it.trim() }.toList(),
+                filesPath = dir,
+                basePath = base,
+                format = SubFormat.valueOf(format)
+            )
+        } catch (exception: Exception) {
+            when (exception) {
+                is NullPointerException -> println("ERROR: All the arguments are required")
```
Comment (Juan-EBI) [inline commons/commons-transpiler/src/main/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLine.kt]: so if an argument is mising `NullPointerException` is throw? it surprice me that library trow such generic error and not information of even the parameter that was mising considering you are flaging parameters as required

Comment (jhoanmanuelms) [reply commons/commons-transpiler/src/main/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLine.kt]: You were absolutely right man, the library does have that option, it's just really hidden in the documentation 😅

File: commons/commons-transpiler/src/main/kotlin/ac/uk/ebi/transpiler/exception/InvalidColumnException.kt (hunk 2 above)
Comment (jhoanmanuelms) [reply commons/commons-transpiler/src/main/kotlin/ac/uk/ebi/transpiler/exception/InvalidColumnException.kt]: Done

File: commons/commons-transpiler/src/test/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLineTest.kt (hunk 1 above)
Comment (jhoanmanuelms) [reply commons/commons-transpiler/src/test/kotlin/ac/uk/ebi/transpiler/cli/TranspilerCommandLineTest.kt]: Done

**Pull request merged**